### PR TITLE
FUSETOOLS-2117 - Improve compatibility with Spring IDE

### PR DESCRIPTION
**Pull request by apupier**
Allows to have Bean code completion in xml

- Fix creation of Spring Templates when Spring IDE is installed
- Fix "Launch as camel context" shortcut when Spring IDE is installed

Can't write tests with current state as we don't have Spring IDE in TP

### CODE REVIEW

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java [hunk 1]
```diff
@@ -89,13 +95,25 @@ private boolean isTestProjectFolder(IProject project, IResource f) {
...
-				&& ifile.getContentDescription()
-							.getContentType()
-							.getId()
-							.equals(CamelUtils.FUSE_CAMEL_CONTENT_TYPE);
+		if( ifile != null
+				&& ifile.isSynchronized(IResource.DEPTH_ZERO)){
+			IContentDescription contentDescription = ifile.getContentDescription();
+			if(contentDescription != null){
+				String contentTypeId = contentDescription.getContentType().getId();
+				if("com.springsource.sts.config.ui.beanConfigFile".equals(contentTypeId)){
```
Comment (bfitzpat) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: Exposed string... Maybe make a constant or  //$NON-NLS-1$?

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java [hunk 2]
```diff
@@ -89,13 +95,25 @@ private boolean isTestProjectFolder(IProject project, IResource f) {
...
+					// Should we filter only for known conflicting content types? If not, it might impact performance.
+					try {
+						ByteArrayInputStream markableAndResettableStream = new ByteArrayInputStream(Files.readAllBytes(ifile.getLocation().toFile().toPath()));
+						return XMLContentDescriber.VALID == new CamelNamespaceXmlContentDescriber().describe(markableAndResettableStream, null);
+					} catch (IOException e) {
+						CamelModelServiceCoreActivator.pluginLog().logInfo("Cannot check Content type of "+ ifile.getName(), e);
```
Comment (bfitzpat) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: I know this is just going into the log, but should we externalize it? Or mark it to not be externalized via  //$NON-NLS-1$?

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java (hunk 1 above)
Comment (apupier) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: fixed

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java (hunk 2 above)
Comment (apupier) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: fixed

**Pull request merged**